.github/workflows/pr-realm-js.yml

```diff
@@ -440,90 +430,110 @@ jobs:
+        id: baas
+        uses: ./.github/actions/baas-test-server
+        with:
+          branch: ${{ env.BAAS_BRANCH }}
+        env:
+          BAASAAS_KEY: ${{ secrets.BAASAAS_KEY }}
+
+      - name: Create Mocha Remote Context
+        id: mocha-env
+        run: echo "context=syncLogLevel=warn,longTimeoutMs=${{ env.LONG_TIMEOUT }},baseUrl=${{ steps.baas.outputs.baas-url }}" >> $GITHUB_OUTPUT
+
+      - name: Run tests
+        env:
+          MOCHA_REMOTE_CONTEXT: ${{ steps.mocha-env.outputs.context }}
+        # The non react native environments should not take so long
+        timeout-minutes: 60
+        run: ${{ env.wrapper }} npm run ${{ matrix.script_name}} --prefix integration-tests/environments/${{ matrix.environment }}
+
+  apple-tests:
+    name: Test iOS (React Native)
+    needs:
+      - generate-jsi
+      - build-ts
+      - prebuild-apple
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_14.3.1.app
+      IOS_DEVICE_NAME: iPhone 14
+    runs-on: macos-latest-large
+    timeout-minutes: 60
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+
+      - name: Setup node version
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: npm
+
+      # - name: Restore React Native cache
+      #   uses: actions/cache@v4
+      #   with:
+      #     path: '**/Pods'
+      #     key: ${{ runner.os }}-react-native-test-app-${{ hashFiles('**/Podfile.lock', './src/**', './vendor/**') }}
+      #     restore-keys: |
+      #       ${{ runner.os }}-react-native-test-app-
 
       - name: MSVC Setup
         if: ${{ runner.os == 'Windows' }}
         uses: ilammy/msvc-dev-cmd@v1
 
       # we use a different version for Android, and it is specified below
       - name: Setup CMake
-        if: ${{ matrix.variant.os != 'android' }}
         uses: jwlawson/actions-setup-cmake@v2
         with:
           cmake-version: ${{ env.CMAKE_VERSION }}
 
       - name: ccache
         uses: hendrikmuhs/ccache-action@v1
-        if: ${{ matrix.variant.environment == 'react-native-test-app' }}
         with:
-          key: ${{ runner.os }}-${{ matrix.variant.os }}-${{ matrix.variant.arch }}
+          key: ${{ github.job }}
           max-size: '2.0G'
+          create-symlink: true
 
-      # in CI file timestamps change with every run so instead rely on file content hashing
-      # https://reactnative.dev/docs/build-speed#using-this-approach-on-a-ci
-      - name: Configure ccache
-        if: ${{ matrix.variant.environment == 'react-native-test-app' }}
-        run: ccache --set-config="compiler_check=content"
-
-      # Hermes doesn't work with Cocoapods 1.15.0
-      # https://forums.developer.apple.com/forums/thread/745518
-      - name: Install older Cocoapods version
-        if: ${{ matrix.variant.os == 'ios' }}
-        uses: maxim-lobanov/setup-cocoapods@v1
-        with:
-          version: 1.14.3
```

Comment:
This got fixed in 1.15.2 which is now installed on the runner and there's no need for the workaround anymore.